src/lib/import_export/csv_parser.cpp

```diff
@@ -13,23 +13,26 @@
 #include "import_export/csv_meta.hpp"
 #include "resolve_type.hpp"
 #include "scheduler/job_task.hpp"
+#include "storage/dictionary_compression.hpp"
 #include "storage/table.hpp"
 #include "utils/assert.hpp"
 #include "utils/load_table.hpp"
 
 namespace opossum {
 
-std::shared_ptr<Table> CsvParser::parse(const std::string& filename, const std::optional<CsvMeta>& csv_meta) {
+std::shared_ptr<Table> CsvParser::parse(const std::string& filename, const std::optional<CsvMeta>& csv_meta,
+                                        const bool auto_compress) {
   // If no meta info is given as a parameter, look for a json file
   if (csv_meta == std::nullopt) {
     _meta = process_csv_meta_file(filename + CsvMeta::META_FILE_EXTENSION);
   } else {
     _meta = csv_meta.value();
   }
-  return _parse(filename);
+
+  return _parse(filename, auto_compress);
 }
 
-std::shared_ptr<Table> CsvParser::_parse(const std::string& filename) {
+std::shared_ptr<Table> CsvParser::_parse(const std::string& filename, const bool auto_compress) {
```

Comment:
What is your reasoning to have `auto_compress` as a parameter here? In my understanding, the result of auto compress is tightly coupled with the chunk size. Shouldn't it be stored together with the chunk size in the meta file?

Reply:
I agree.

Reply:
Yes you are right. I had it as parameter since I though it would not be directly coupled to the content of the CSV, but then again, it IS coupled with the chunk_size which is also in the meta file. `auto_compress` is in the meta file as well now.